src/index.js

```diff
@@ -265,6 +298,30 @@ export default function fetch(url, opts) {
 
 };
 
+function fixResponseChunkedTransferBadEnding(request, errorCallback) {
+	let socket;
+
+	request.on('socket', s => {
+		socket = s;
+	});
+
+	request.on('response', response => {
+		const {headers} = response;
+		if (headers['transfer-encoding'] === 'chunked' && !headers['content-length']) {
+			socket.addListener('close', () => {
+				// if a data listener is still present we didn't end cleanly
+				const hasDataListener = socket.listenerCount('data') > 0;
```

Comment:
In #1064 this checks the incoming bytes to make sure it ends with `0\r\n` which signifies the last chunk is received.  I found this didn't work well against real-world servers as it assumes the `'data'` event would align with incoming chunks whereas I don't think there's any such guarantee.

I've reused the same check as the [patch for node 10-12 above](https://github.com/node-fetch/node-fetch/pull/1172/files#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556R122) which seems to work well.

Reply:
@achingbrain The data event doesn't have to align; this check is only applicable at the very end when the socket is closed and the last bytes of data are pushed out to the data event.  The only way for a chunked transfer to indicate "the end" is for `0\r\n` to be the last 3 bytes at this point, which is what the check was for.

Reply:
This has caused problems when reusing sockets for multiple requests, as when you use an `https.Agent` in ‘keep-alive’ mode.

https://github.com/node-fetch/node-fetch/issues/1767